parsl/executors/workqueue/workqueue_worker.py

```diff
@@ -59,15 +57,19 @@ def check_file(parsl_file_obj, mapping, file_type_string):
         input_function = open(input_function_file, "rb")
         function_info = pickle.load(input_function)
         # Extract information from transferred source code
-        if source:
+        if "source code" in function_info:
+            fn_from_source = True
             source_code = function_info["source code"]
             name = function_info["name"]
             args = function_info["args"]
             kwargs = function_info["kwargs"]
         # Extract information from function pointer
-        else:
+        elif "byte code" in function_info:
+            fn_from_source = False
             from ipyparallel.serialize import unpack_apply_message
-            func, args, kwargs = unpack_apply_message(function_info, user_ns, copy=False)
+            func, args, kwargs = unpack_apply_message(function_info["byte code"], user_ns, copy=False)
+        else:
+            raise Exception("Function file does not have a valid function representation.")
```

Comment:
If the function fails at this point, how does the error notification propagate back to Parsl ?

Reply:
I don't know if they do with the current version. I think that all exceptions should be reworked to match what parsl expects. Right now there is generic wq error exception, but as in this code snippet, this errors may be better communicated to parsl. Let me open an issue for that.

Reply:
See #1688